src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionReader.cs

```diff
@@ -279,6 +287,16 @@ private object ReadItemOfPrimitiveType(XmlReaderDelegator xmlReader, XmlObjectSe
             return value;
         }
 
+        private static object ReadISerializable(XmlReaderDelegator xmlReader, XmlObjectSerializerReadContext context, ClassDataContract classContract)
+        {
+            object obj;
+            SerializationInfo serializationInfo = context.ReadSerializationInfo(xmlReader, classContract.UnderlyingType);
+            StreamingContext streamingContext = context.GetStreamingContext();
+            ConstructorInfo iSerializableConstructor = classContract.GetISerializableConstructor();
+            obj = iSerializableConstructor.Invoke(new object[] { serializationInfo, streamingContext });
```

Comment:
Could a stronger typed delegate be returned instead? All ISerializable constructors have the same parameters.